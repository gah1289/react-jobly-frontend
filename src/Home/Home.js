import React, { useContext } from 'react';
import { Spinner } from 'reactstrap';
import ItemContext from '../ItemContext';
import { Button } from 'reactstrap';

function Home() {
	const { isLoggedIn, user, isLoading } = useContext(ItemContext);

	if (!isLoggedIn) {
		return (
			<div>
				<h1>Welcome! </h1>
				<a href="/login">
					<Button>Login</Button>
				</a>

				<a href="/signup">
					<Button>Signup</Button>
				</a>
			</div>
		);
	}
	else {
		if (isLoading) {
			return <Spinner />;
		}
		else {
			return (
				<div>
					<h1>Hello, {user.current.firstName}!</h1>
				</div>
			);
		}
	}
}

// if (isLoading) {
// 	return <Spinner />;
// }
// else {
// if (user.current.firstName) {
// 	return (
// 		<div className="Home">
// 			{!isLoggedIn ? (
// 				<div>
// 					<h1>Welcome! </h1>
// 					<a href="/login">
// 						<Button>Login</Button>
// 					</a>

// 					<a href="/signup">
// 						<Button>Signup</Button>
// 					</a>
// 				</div>
// 			) : (
// 				<div>
// 					<h1>Hello, {user.current.firstName}!</h1>
// 				</div>
// 			)}
// 		</div>
// 	);
// }
// }
// }

export default Home;
