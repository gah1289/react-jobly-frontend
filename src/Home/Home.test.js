import { render, screen } from '@testing-library/react';
import Home from './Home';

import ItemContext from '../ItemContext';

describe('Loads Home Page', function() {
	let isLoggedIn;
	let user;
	let isLoading;
	let HomeComponent;
	let testUser = {
		username  : 'testUser',
		firstName : 'Test',
		lastName  : 'User',
		email     : 'test@user.com',
		password  : 'password'
	};

	it('renders without crashing', function() {
		HomeComponent = (
			<ItemContext.Provider value={((isLoggedIn = false), (user = undefined), (isLoading = false))}>
				{' '}
				<Home />{' '}
			</ItemContext.Provider>
		);
		render(HomeComponent);
	});

	it('prompts user to log in', function() {
		HomeComponent = (
			<ItemContext.Provider value={((isLoggedIn = false), (user = undefined), (isLoading = false))}>
				{' '}
				<Home />{' '}
			</ItemContext.Provider>
		);
		const { asFragment } = render(HomeComponent);
		expect(asFragment()).toMatchSnapshot();
		const welcomeBanner = screen.queryByText(/Welcome/i);
		expect(welcomeBanner).toBeInTheDocument();
		const login = screen.queryByText(/Login/i);
		expect(login).toBeInTheDocument();
	});
	it('greets user once they have logged in', function() {
		user = testUser;
		user.current = testUser;
		isLoggedIn = true;
		isLoading = false;
		HomeComponent = (
			<ItemContext.Provider value={{ user, isLoggedIn, isLoading }}>
				{' '}
				<Home />{' '}
			</ItemContext.Provider>
		);
		const { asFragment } = render(HomeComponent);
		expect(asFragment()).toMatchSnapshot();
		const greetUserByName = screen.queryByText(/Hello, Test!/i);
		expect(greetUserByName).toBeInTheDocument();
	});
});
