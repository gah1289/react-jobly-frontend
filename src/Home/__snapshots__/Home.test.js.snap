// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads Home Page greets user once they have logged in 1`] = `
<DocumentFragment>
   
  <div>
    <h1>
      Hello, Test!
    </h1>
  </div>
   
</DocumentFragment>
`;

exports[`Loads Home Page prompts user to log in 1`] = `
<DocumentFragment>
   
  <div>
    <h1>
      Welcome! 
    </h1>
    <a
      href="/login"
    >
      <button
        class="btn btn-secondary"
      >
        Login
      </button>
    </a>
    <a
      href="/signup"
    >
      <button
        class="btn btn-secondary"
      >
        Signup
      </button>
    </a>
  </div>
   
</DocumentFragment>
`;
